Assets/HoloToolkit/Input/Scripts/Microphone/DictationInputManager.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Text;
+using UnityEngine;
+using UnityEngine.Windows.Speech;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// Singleton class that implements the DictationRecognizer to convert the user's speech to text.
+    /// The DictationRecognizer exposes dictation functionality and supports registering and listening for hypothesis and phrase completed events.
+    /// </summary>
+    public class DictationInputManager : Singleton<DictationInputManager>, IInputSource
+    {
+        /// <summary>
+        /// Initial value for InitialSilenceTimeout.
+        /// </summary>
+        private static float initialSilenceTimeout = 5f;
+
+        /// <summary>
+        /// The time length in seconds before dictation recognizer session ends due to lack of audio input in case there was no audio heard in the current session.
+        /// </summary>
+        public static float InitialSilenceTimeout
```

Comment:
Why are InitialSilenceTimeout, AutoSilenceTimerout, and RecordingTime static? Since DictationInputManager is a singleton, would it be more natural for these to be ordinary public fields referenced via DictationInputManager.Instance.InitialSilenceTimeout? Also, it could be more convenient to specify them on the manager instead of duplicating these fields on items such as DictationRecordButton that have their own local settings that are transferred to the global settings.